drivers/can/can_stm32.c

```diff
@@ -468,11 +514,24 @@ int can_stm32_send(struct device *dev, const struct zcan_frame *msg,
 	k_mutex_unlock(tx_mutex);
 
 	if (callback == NULL) {
-		k_sem_take(&mb->tx_int_sem, K_FOREVER);
-		return mb->error_flags;
+		k_sem_init(&mb->tx_int_sem, 0, 1);
```

Comment:
Should you really initialize the semaphore here? What happens if the transmission already happened before this point in time and the tx interrupt has already fired?

Reply:
mb semaphores are removed now.